sdk/core/Azure.Core/src/Internal/RehydrationOperation.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    internal class RehydrationOperation : Operation
+    {
+        private readonly NextLinkOperationImplementation _nextLinkOperation;
+        private readonly OperationInternal _operation;
+
+        public RehydrationOperation(NextLinkOperationImplementation nextLinkOperation, OperationState operationState, ClientOptions? options = null)
+        {
+            _nextLinkOperation = nextLinkOperation;
+            _operation = operationState.HasCompleted
+                ? _operation = new OperationInternal(operationState)
+                : new OperationInternal(_nextLinkOperation, new ClientDiagnostics(options ?? ClientOptions.Default), operationState.RawResponse);
+        }
+
+        public override string Id => _nextLinkOperation.OperationId ?? string.Empty;
```

Comment:
since `_nextLinkOperation.OperationId` is nullable, we need to handle it properly here.
There are 3 options:
- Make `Operation.Id` nullable, this should an acceptable breaking change
- return `string.Emepty` as implemented
- `public override string Id => _nextLinkOperation.OperationId!` to fool the compiler

Reply:
We would want to evaluate whether making `Operation.Id` nullable is the right design for this type.  My understanding is that it is not -- it is a fundamental property of `Operation` that is required for using the type.  I think it would make sense to take a little time to understand what flows `Operation.Id` is used in, and what and end-user would do in the case that it is null.  We would want to weigh that against the case you have where it is required to be null -- should we throw when it is null instead of returning null, e.g.?  Is it a valid scenario that users of `Operation` need to be aware of and handle in the main use-cases?  These would be questions we would need to answer concretely before making such a change from this API, and we would need to provide documentation to users so they can change their apps to handle it.  Starting to return null from a non-nullable API is a pretty big breaking change that would require a lot of management -- we have had a number of problems in other areas of Core due to the fact that we do this sometimes, and customers have complained about it.

Reply:
> Make Operation.Id nullable, this should be an acceptable breaking change

To state the obvious, we do not make breaking changes under normal circumstances.   While there are exceptions, it is not something that is taken lightly nor that would be done without strong justification and discussion/approval from the architects.   Can you please share what rationale justifies this as "an acceptable" breaking change and which of our .NET architects signed-off on that decision?

Reply:
Make `OperationId` not nullable and set it default to "NOT_SET"